quarkus create -S 2.5 fails if registry.quarkus.redhat.com is first

### Describe the bug

I have quarkus.redhat.com and quarkus.io listed:
```
quarkus registry list
Configured Quarkus extension registries:
playground.io (disabled)
registry.quarkus.redhat.com
registry.quarkus.io
(Read from /Users/max/.quarkus/config.yaml)
```

but running create using -S 2.5 (latest recommend community) fails:

```
quarkus create -S 2.5
Creating an app (default project type, see --help).
[ERROR] ❗  Unable to create project: Failed to locate stream 2.5 in platform(s): com.redhat.quarkus.platform, io.quarkus.platform
```

If I run `quarkus registry remove registry.quarkus.redhat.com` then `quarkus create -S 2.5` starts working 

### Expected behavior

should generate for stream 2.5 as found in the registry.quarkus.io registry.

Or at least state which platforms was searched so user can understand why it was rejected/not found.

### Actual behavior

just says it cannot find 2.5 even though it is there.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/project/create/UpstreamStreamCoordsWithNullPlatformKeyTest.java
+++ b/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/project/create/UpstreamStreamCoordsWithNullPlatformKeyTest.java
@@ -1,0 +1,76 @@
+package io.quarkus.devtools.project.create;
+
+import io.quarkus.devtools.testing.registry.client.TestRegistryClientBuilder;
+import io.quarkus.maven.ArtifactCoords;
+import io.quarkus.registry.catalog.PlatformStreamCoords;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class UpstreamStreamCoordsWithNullPlatformKeyTest extends MultiplePlatformBomsTestBase {
+
+    private static final String DOWNSTREAM_PLATFORM_KEY = "io.downstream.platform";
+    private static final String UPSTREAM_PLATFORM_KEY = "io.upstream.platform";
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        TestRegistryClientBuilder.newInstance()
+                //.debug()
+                .baseDir(configDir())
+                // registry
+                .newRegistry("downstream.registry.test")
+                .recognizedQuarkusVersions("*-downstream")
+                // platform key
+                .newPlatform(DOWNSTREAM_PLATFORM_KEY)
+                .newStream("1.0")
+                // 1.0.4 release
+                .newRelease("1.1.1-downstream")
+                .quarkusVersion("1.1.1-downstream")
+                .upstreamQuarkusVersion("1.1.1")
+                // default bom including quarkus-core + essential metadata
+                .addCoreMember()
+                // foo platform member
+                .newMember("acme-a-bom").addExtension("io.acme", "ext-a", "1.1.1-downstream").release()
+                .stream().platform().registry()
+                .clientBuilder()
+                .newRegistry("upstream.registry.test")
+                // platform key
+                .newPlatform(UPSTREAM_PLATFORM_KEY)
+                // 2.0 STREAM
+                .newStream("2.0")
+                // 2.0.5 release
+                .newRelease("2.0.5")
+                .quarkusVersion("2.0.5")
+                // default bom including quarkus-core + essential metadata
+                .addCoreMember()
+                .newMember("acme-a-bom").addExtension("io.acme", "ext-a", "2.0.5").release().stream().platform()
+                // 1.0 STREAM
+                .newStream("1.0")
+                .newRelease("1.1.1")
+                .quarkusVersion("1.1.1")
+                // default bom including quarkus-core + essential metadata
+                .addCoreMember()
+                .newMember("acme-a-bom").addExtension("io.acme", "ext-a", "1.1.1")
+                .registry()
+                .clientBuilder()
+                .build();
+
+        enableRegistryClient();
+    }
+
+    protected String getMainPlatformKey() {
+        return UPSTREAM_PLATFORM_KEY;
+    }
+
+    @Test
+    public void createUsingStream2_0() throws Exception {
+        final Path projectDir = newProjectDir("created-using-upstream-2.0");
+        createProject(projectDir, new PlatformStreamCoords(null, "2.0"), List.of("ext-a"));
+
+        assertModel(projectDir,
+                List.of(mainPlatformBom(), platformMemberBomCoords("acme-a-bom")),
+                List.of(new ArtifactCoords("io.acme", "ext-a", null)),
+                "2.0.5");
+    }
+}

--- a/independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/ExtensionCatalogResolver.java
+++ b/independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/ExtensionCatalogResolver.java
@@ -489,7 +489,7 @@ public class ExtensionCatalogResolver {
 
         PlatformStream stream = null;
         int registryIndex = 0;
-        while (registryIndex < registries.size()) {
+        while (registryIndex < registries.size() && stream == null) {
             final RegistryExtensionResolver qer = registries.get(registryIndex++);
             final PlatformCatalog platforms = qer.resolvePlatformCatalog();
             if (platforms == null) {
@@ -508,8 +508,10 @@ public class ExtensionCatalogResolver {
                     continue;
                 }
                 stream = platform.getStream(streamId);
+                if (stream != null) {
+                    break;
+                }
             }
-            break;
         }
 
         if (stream == null) {

